lints/lint_dnsname_contains_question_marks.go

```diff
@@ -0,0 +1,46 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"strings"
+)
+
+type DNSNameContainsQuestionMarks struct{}
+
+func (l *DNSNameContainsQuestionMarks) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameContainsQuestionMarks) CheckApplies(c *x509.Certificate) bool {
+	return true
+}
+
+func isPrependedByQuestionMarks(domain string) bool {
+	if strings.HasPrefix(domain, "?.") {
+		return true
+	}
+	return false
+}
+
+func (l *DNSNameContainsQuestionMarks) Execute(c *x509.Certificate) *LintResult {
```

Comment:
This should probably only run if `cn != ""`

Reply:
This really only checks if this is prepended by _a_ question mark, not multiple question marks.

Reply:
Yeah, I think that encompasses everything we want to check though? `?.?.abc.com` is still prepended with `?.` Maybe I'm not understanding this comment

Reply:
The thing should really just be named "IsRedactedCertificate"